Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -901,6 +908,34 @@ void AODProducerWorkflowDPL::init(InitContext& ic)
   // Needed by MCH track extrapolation
   o2::base::GeometryManager::loadGeometry();
 
+  // writing metadata if it's not yet in AOD file
+  // note: `--aod-writer-resmode "UPDATE"` have to be used,
+  //       so that metadata is not overwritten
+  mResFile += ".root";
+  auto* fResFile = TFile::Open(mResFile, "UPDATE");
+  if (fResFile) {
+    if (!fResFile->FindObjectAny("metaData")) {
+      // populating metadata map
+      TString dataType = mUseMC ? "MC" : "RAW";
+      mMetaData.Add(new TObjString("DataType"), new TObjString(dataType));
+      mMetaData.Add(new TObjString("Run"), new TObjString("3"));
+      TString converterVersion = "o2 ";
+      converterVersion += o2::fullVersion();
+      converterVersion += " ; root ";
+      converterVersion += ROOT_RELEASE;
+      mMetaData.Add(new TObjString("Run3ConverterVersion"), new TObjString(converterVersion));
+      mMetaData.Add(new TObjString("RecoPassName"), new TObjString(mRecoPass));
+      mMetaData.Add(new TObjString("AnchorProduction"), new TObjString(mAnchorProd));
+      mMetaData.Add(new TObjString("AnchorPassName"), new TObjString(mAnchorPass));
+      mMetaData.Add(new TObjString("LPMProductionTag"), new TObjString(mLPMProdTag));
+      LOGF(info, "Metadata: writing into %s", mResFile);
+      fResFile->WriteObject(&mMetaData, "metaData");
+    } else {
+      LOGF(info, "Metadata: target file not found or metadata is already written");
```

Comment:
I think this is not the case of file not found. It is just the if  (!fResFile->FindObjectAny("metaData"))
Are there valid cases when this is allowed? Otherwise it should be `fatal`
I think it should also be `fatal` if the output file is not found